action.yaml

```diff
@@ -52,11 +52,33 @@ runs:
       shell: bash
       run: |
         cd $(mktemp -d)
-        wget --quiet -U "GitHub setup-chainctl" -O chainctl "https://dl.${{ inputs.environment }}/chainctl/latest/chainctl_linux_$(uname -m)"
+
+        # Massage GitHub's values to the ones we expect.
+        # https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context
+        os=$(echo "${{ runner.os }}" | tr '[:upper:]' '[:lower:]')
+        if [[ "${os}" == "macos" ]]; then
+          os="darwin"
+        fi
+
+        arch="${{ runner.arch }}"
+        if [[ "${arch}" == "X64" ]]; then
+          arch="amd64"
+        elif [[ "${arch}" == "ARM64" ]]; then
+          arch="arm64"
+        fi
+
+        url="https://dl.${{ inputs.environment }}/chainctl/latest/chainctl_${os}_${arch}"
+        if [[ "${os}" == "windows" ]]; then
+          url="${url}.exe"
+        fi
+        echo "Downloading chainctl from ${url}"
+
+        curl -o ./chainctl -fsL "${url}"
+        chmod +x ./chainctl
         CHAINCTL_INSTALL_PATH="${HOME}/.local/bin"
+        mv ./chainctl "${CHAINCTL_INSTALL_PATH}"
         # Ensure install directory is on the PATH for future steps
         echo "${CHAINCTL_INSTALL_PATH}" >> "${GITHUB_PATH}"
-        install -D --mode 0555 ./chainctl --target-directory "${CHAINCTL_INSTALL_PATH}"
```

Comment:
Windows (and possibly macOS) runners don't have `install`, but it's just `chmod && mv` anyway